src/Pages/Signup/Signup.jsx

```diff
@@ -0,0 +1,163 @@
+import "../Signup/Signup.css";
+import { Link, useNavigate } from "react-router-dom";
+import { Header } from "../../components/Header/Header";
+import { useState } from "react";
+import { useAuth } from "../../Context/auth-context";
+import axios from "axios";
+
+export const Signup = () => {
+  const [user, setUser] = useState({
+    firstName: "",
+    lastName: "",
+    email: "",
+    password: "",
+  });
+
+  const [checked, setChecked] = useState(false);
+  const [showPassword, setShowPassword] = useState("password");
+  const [icon, setIcon] = useState("visibility_off");
+  const [errorMsg,setErrorMsg]=useState(false)
+  const [color,setColor]=useState("")
+  const navigate = useNavigate();
+  const { setAuth } = useAuth();
+
+  const userInputHandler = (e) => {
+    const { name, value } = e.target;
+    setUser({ ...user, [name]: value });
+    
+    setErrorMsg(false)
+  };
+
+
+  const signupHandler = async () => {
+    if (
+      !user.firstName ||
+      !user.lastName ||
+      !user.email ||
+      !user.password ||
+      !checked
+    ) {
+      setErrorMsg(true);
+      setColor("red");
+      return;
+    }
+    try {
+      const response = await axios.post("/api/auth/signup", {
+        firstName: user.firstName,
+        lastName: user.lastName,
+        email: user.email,
+        password: user.password,
+      });
+
+      localStorage.setItem("token", response.data.encodedToken);
+
+      setAuth(true);
+      navigate("/login");
+    } catch (error) {
+      console.log(error);
```

Comment:
you can show error in an  alert if possible

Reply:
ok